### DISCUSSION TITLE
Example of how to use the the GraphQlService inside an extension

### DISCUSSION THREAD

**Discussion by MichaelFBA** [directus/directus#13931] [Archive]
Could you provide an example of how to use the graphqlservice so it can be used inside of an extension

```
const graphQl = new GraphQLService({
        accountability,
        schema,
        scope: 'items'
      });
      
      const graphQlResult = await graphQl.execute({
        document: `query {
          shopping_cart{
            id
          }
        }`
      });
 ```
 This example throws an error about validation so I suspect I am missing something

**Comment by licitdev** [comment]
The `document` has to be parsed and the `contextValue` should be provided.

```
import { parse, Source } from 'graphql';

...

const query = `query {
  shopping_cart {
    id
  }
}`;

let document;

try {
  document = parse(new Source(query));
} catch (err) {
  throw new Error(err);
}

const graphQlResult = await graphQl.execute({
  document,
  contextValue: { req, res },
});

```
Please refer to https://github.com/directus/directus/blob/main/api/src/middleware/graphql.ts for details.

**Reply by benhaynes** [reply]
CC @erondpowell

**Reply by SergeyRe** [reply]
i get: 
[RollupError] Unexpected token (Note that you need plugins to import files that are not JavaScript)
 on build it.
 Please provide full example for reference.Thank you

**Reply by SergeyRe** [reply]
it was graphql@16.8.0 specific bug - see [here](https://github.com/graphql/graphql-js/issues/3928)
Solved by  using 16.6.0
